assemblies/resources/bin/check_ports

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Licensed to The Apereo Foundation under one or more contributor license
+# agreements. See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.
+#
+#
+# The Apereo Foundation licenses this file to you under the Educational
+# Community License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License
+# at:
+#
+#   http://opensource.org/licenses/ecl2.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# -----------------------------------------------------------------------
+# check if http or https port is available to run opencast otherwise exit
+
+if [ -d "$1" ]; then
+  SETENV_ETC="$1"
+else
+  if [ -d "${KARAF_BASE}" ]; then
+    SETENV_ETC="${KARAF_BASE}/etc"
+  else
+    SETENV_ETC="etc"
+  fi
+fi
+
+check_port() {
+  if [ "$1" ] ; then
+    hexPort=$(echo "ibase=10; obase=16; $1" | bc)
+    for i in /proc/net/*; do 
+	if  [ -f "$i" ] && [ -r "$i" ]; then
+	   o=$(grep -c ":$hexPort " "$i")
+	   if [ "$o" = "1" ]; then
+		return 1;
+	   fi
+	fi
+    done
+  fi
+  return 0;
+}
```

Comment:
Could you convert tabs to spaces (or vice versa).

Reply:
will be spaces in the end